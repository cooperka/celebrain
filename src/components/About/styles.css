.About {
  min-height: 300px;
  margin: 4em 0;
  display: flex;
  flex-direction: column;
  font-weight: 500;
}

.About .title {
  margin: 1rem;
  font-size: 5em;
}

.About .subtitle {
  margin: 1rem auto 0 auto;
  font-size: 2.2em;
  max-width: 20em;
}

.About .section {
  margin: 1em auto;
  font-size: 1.4em;
  max-width: 40em;
}

.About .section .header {
  margin-bottom: 0.5em;
}

.About .section .description {
  font-weight: normal;
}
