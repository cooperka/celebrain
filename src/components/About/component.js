// @flow

import React from 'react';

import './styles.css';

function About() {
  return (
    <div className="About">
      <div className="title">About</div>

      <div className="section">
        <div className="subtitle">What does LSR mean?</div>

        <div className="header">L - Lovable</div>
        <div className="description">People should love using your stuff.</div>

        <div className="header">S - Simple</div>
        <div className="description">Your stuff should be understandable and straightforward.</div>

        <div className="header">R - Refined</div>
        <div className="description">Your stuff should improve over time; it{"'"}s probably not perfect yet.</div>
      </div>

      <div className="section">
        <div className="subtitle">Why those three words?</div>
        <div className="description">
          LSR comes from a combination of two powerful concepts in software development:{' '}
          <a href="http://blog.crisp.se/2016/01/25/henrikkniberg/making-sense-of-mvp">
            Minimal Lovable Product
          </a>
          {' + '}
          <a href="https://steveblank.com/2015/05/06/build-measure-learn-throw-things-against-the-wall-and-see-if-they-work/">
            Build Measure Learn
          </a>
          . I highly recommend reading through those links to know more.
        </div>
      </div>
    </div>
  );
}

export default About;
