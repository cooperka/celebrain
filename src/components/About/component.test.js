// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import About from './component';

it('renders', () => {
  const tree = renderer.create(<About />);
  expect(tree.toJSON()).toMatchSnapshot();
});
