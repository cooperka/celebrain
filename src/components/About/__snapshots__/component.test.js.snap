// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="About page"
>
  <div
    className="title"
  >
    About
  </div>
  <div
    className="section"
  >
    <div
      className="subtitle"
    >
      Inspiration
    </div>
    <div
      className="description"
    >
      I
      '
      ve always had a difficult time remembering peoples names -- especially movie characters and celebrities -- so I
      '
      ve recently been trying harder to improve. This game is a fun attempt at solving two problems at once.
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="subtitle"
    >
      Which celebrities?
    </div>
    <div
      className="description"
    >
      Currently, the pool of celebrities comes from the Wikipedia categories for American
       
      <a
        href="https://en.wikipedia.org/wiki/Category:American_male_film_actors"
      >
        actors
      </a>
       and 
      <a
        href="https://en.wikipedia.org/wiki/Category:American_film_actresses"
      >
        actresses
      </a>
      . It includes every person in each of those categories for whom a picture is available -- about 5300 men and 4200 women. This will be expanded soon.
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="subtitle"
    >
      How is popularity determined?
    </div>
    <div
      className="description"
    >
      The 
      <a
        href="https://www.mediawiki.org/wiki/API:Main_page"
      >
        Wikimedia API
      </a>
       provides monthly Wikipedia pageview metrics which I used to determine popularity.
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="subtitle"
    >
      Attribution
    </div>
    <div
      className="description"
    >
      Images are sourced from Wikipedia and may be subject to copyright. Click on the images to find their original source and licensing on the Wikimedia Commons.
    </div>
  </div>
</div>
`;
